src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1723,7 +1727,7 @@ private BoundExpression ConvertSwitchExpression(BoundUnconvertedSwitchExpression
 
                 return new BoundConversion(
                     syntax,
-                    source,
+                    BindToNaturalType(source, diagnostics),
```

Comment:
Should we be calling `BindToTypeForErrorRecovery` instead? It seems like the main difference is around error reporting; any errors from `BindToNaturalType` will be put into `diagnostics`, while `BindToTypeForErrorRecovery` will discard them. Do we already have a diagnostic at this point?

Reply:
We will definitely have *some* diagnostic about this expression, because of the preceding `if (!hasErrors) GenerateImplicitConversionError(...)`.

re: the difference in practice. If we used BindToTypeForErrorRecovery here, then test `Repro_VsFeedback_2088611` wouldn't report `ERR_ImplicitObjectCreationNoTargetType`. It would only report `ERR_ConversionNotTupleCompatible`. I feel like in this particular case, it's maybe better to have a squiggle on the new() because it happens to be related to the ambiguity. But, ideally, we would be able to report some kind of user-defined operator overload resolution error here, I think.